src/nvim/optionstr.c

```diff
@@ -1698,6 +1698,9 @@ const char *did_set_iconstring(optset_T *args)
 /// The 'inccommand' option is changed.
 const char *did_set_inccommand(optset_T *args FUNC_ATTR_UNUSED)
 {
+  if (cmdpreview) {
+    return e_invarg;
+  }
```

Comment:
Maybe we could allow using `set inccommand` if the value has not changed? So perhaps something like:
```suggestion
  if (cmdpreview && strcmp(args->os_oldval.string, args->os_newval.string) != 0) {
    return e_invarg;
  }
```
<br />
Unsure if this is worth it though.

Reply:
I don't think it's worth adding any additional logic here, it just muddies the waters without providing any value. Keep it simple.